File: DemoObjectiveC/DemoObjectiveC/ViewController.m
//
//  ViewController.m
//  DemoObjectiveC
//
//  Created by Carlos Pages on 03/02/2015.
//  Copyright (c) 2015 cepages. All rights reserved.
//

#import "ViewController.h"
#import "DemoObjectiveC-Swift.h"

static NSInteger INITIAL_REMAIN_TIME = 10;


@interface ViewController ()

@property(nonatomic,strong)NSTimer *timer;
@property(nonatomic,assign)NSInteger remainTime;

@property (weak, nonatomic) IBOutlet UIButton *simpleHUDButton;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)activateSimpleHUDToucheIn:(id)sender {
    
    
    CPSimpleHUD *hud = [CPSimpleHUD shareWaitingView];
    hud.loadingLabel.text = @"Simple HUD";
    hud.heightDarkViewContraint.constant = 250;
    hud.widthDarkViewContraint.constant = 250;
    
    [hud show];
    
    
    self.remainTime = INITIAL_REMAIN_TIME;
    
    self.timer = [NSTimer scheduledTimerWithTimeInterval:1 target:self selector:@selector(timerFire:) userInfo:nil repeats:YES];
}


-(void)timerFire:(NSTimer *)timer
{
    self.remainTime--;
    
    [self.simpleHUDButton setTitle:[NSString stringWithFormat:@"Time remaining %li",(long)self.remainTime] forState:UIControlStateNormal];

    if (self.remainTime == 0) {
        [[CPSimpleHUD shareWaitingView] hide];
        [self.simpleHUDButton setTitle:@"Activate Simple HUD" forState:UIControlStateNormal];
        
        [self.timer invalidate];
        self.timer = nil;
    }
    
    //We update the button label
}

@end

File: Demo/SimpleHUD/ViewController.swift
//
//  ViewController.swift
//  SimpleHUD
//
//  Created by Carlos Pages on 28/08/2014.
//  Copyright (c) 2014 Carlos Pages. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    let INITIAL_REMAIN_TIME = 10

    
    @IBOutlet weak var simpleHUDButton: UIButton!
    
    var remainTime:Int = 0
    var timer:NSTimer?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


    @IBAction func activateSimpleHUDToucheIn(sender: AnyObject) {
        
        let buttonTouched = sender as UIButton
        
        let hud = CPSimpleHUD.shareWaitingView
        hud.loadingLabel.text = "Simple HUD"
        hud.heightDarkViewContraint.constant = 250;
        hud.widthDarkViewContraint.constant = 250;
        hud.waitingMode = .SmallCubesBorders
        hud.show()
        

        self.remainTime = INITIAL_REMAIN_TIME
        
        self.timer = NSTimer.scheduledTimerWithTimeInterval(1, target: self, selector: Selector("timerFire:"), userInfo: nil, repeats: true)
        
    }
    
    func timerFire(timer: NSTimer){
        self.remainTime--
        //We update the button label
        self.simpleHUDButton.setTitle("Time remaining \(self.remainTime)", forState:.Normal)
       
        if self.remainTime == 0 {
            CPSimpleHUD.shareWaitingView.hide()
            self.simpleHUDButton.setTitle("Activate Simple HUD", forState:.Normal)

            timer.invalidate()
        }
    }
    
    
}

